\documentclass[10pt,xcolor=dvipsnames]{beamer}
\setbeamertemplate{navigation symbols}{}



\usepackage{color}
\usepackage{CREAL_slides}
\usepackage[latin1]{inputenc} 
\usepackage{calc} 
\usepackage[loadonly]{enumitem}
\usepackage{float}
\usepackage[position=top,singlelinecheck=off]{subfig}

\DeclareMathOperator{\argmax}{arg\,max} 


<<include=FALSE>>=
library(knitr)
opts_chunk$set(
tidy=FALSE, size='footnotesize', warning=FALSE, message=FALSE, fig.align='center', out.width='2in')
@




\title[Omic data analysis]{Multi-omic data analysis}
\author[Juan R Gonzalez]{Juan R Gonzalez}
\institute[CREAL]{BRGE - Bioinformatics Research Group in Epidemiology \\
		  Barcelona Institute for Global Health (ISGlobal) \\ 
		           {\tt e-mail:juanr.gonzalez@isglobal.org} \\
                  \url{http://www.creal.cat/brge} 
                  }
                 
\date{}


\begin{document}

                            
<<setup, echo=FALSE>>=
options(width = 80)
@ 

\frame{\titlepage}

\frame{\tableofcontents}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{frame}\frametitle{Introduction}

\begin{itemize}
\item Undersanding the genetic basis of \textcolor{OliveGreen}{complex traits} is an open question for may researchers
\pause
\item Recent advances in technology has made this problem even more complex by incorporating other pieces
of information such as neuroimaging, daily measurements of enviromental exposures among others
\pause
\item The main challenge is to analyze the vast amount of data that is being generated, particularly how to integrate information from different tables
\end{itemize}
\end{frame}



\begin{frame}\frametitle{Introduction}

\begin{figure}
\begin{center}
 \includegraphics[height=6cm, width=6.5cm]{figures/general_omic_data.png}
\end{center}
\end{figure}

\end{frame}


\begin{frame}\frametitle{Introduction}

\begin{figure}
\begin{center}
 \includegraphics[height=6cm, width=6cm]{figures/supervised_multi_omic.png}
\end{center}
\end{figure}

\end{frame}


%\begin{frame}\frametitle{Introduction}
%\begin{figure}
%\begin{center}
% \includegraphics[height=6cm, width=5cm]{figures/non_supervised_multi_omic.png}
%\end{center}
%\end{figure}
%\end{frame}


\begin{frame}\frametitle{Introduction}

\begin{itemize}
\item Data integration (Integrative bioinformatics, integrated 
analysis, crossomics, multi-dataset analysis, data fusion, ...) is being crucial in Bioinformatics/Biology/Epidemiology
\item Data integration may refer to different aspects
  \begin{itemize}
   \item Computational combination of data (sets)
   \item Simultaneous analysis of different variables from different tables, different time points, different tissues, ...
   \item Provide biological insights by using information from existing databases (ENCODE, GTEX, KEGG, ...)
  \end{itemize}
\item Here we mean the process by which different types of omic data are combined as predictor 
variables to allow more thorough and comprehensive modelling of complex traits or phenotypes
\end{itemize}
\end{frame}


\begin{frame}\frametitle{Introduction}

\begin{itemize}
\item \textcolor{OliveGreen}{Multi-staged analysis}: Involves integrating information using a stepwise or hierarchical analysis approach
\item \textcolor{OliveGreen}{Meta-dimensional analysis}: Refers to the concept of integrating multiple different data types to build a multivariate model associated with a given outcome
\end{itemize}
\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Multi-staged analysis}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{frame}\frametitle{Multi-staged analysis}

\begin{itemize}
\item \textcolor{OliveGreen}{Multi-staged analysis}: Involves integrating information using a stepwise or hierarchical analysis approach
\item \textcolor{Gray}{Meta-dimensional analysis: Refers to the concept of integrating multiple different data types to build a multivariate model associated with a given outcome}
\end{itemize}
\end{frame}


\begin{frame}\frametitle{Multi-staged analysis}
\begin{itemize}
\item \textcolor{OliveGreen}{Genomic variation analysis approaches}: Triangle approach
   \begin{enumerate}
    \item SNPs are associated with the phenotype of interesed and filtered based on p-values
    \item Significant SNPs are tested for association with another level of omic data (i.e eQTLs)
    \item Omic data used in step 2 are then tested for correlation with phenotype of interest
   \end{enumerate}

\pause

\item \textcolor{OliveGreen}{Domain knowledge-guided approaches}: Integration of functional and pathway information
   \begin{enumerate}
    \item Significant results (regions of interest) are annotated in databases such as ENCODE, Roadmap Epigenomic Project, KEGG, ...
    \item We an determine whether those regions are within pathways and/or overlaping with functional units, sucha as transcription factors binding, hyper- or hypo-methylated regions
    \item Overlaping regions are then correlated with the phenotype of interest
    \item The main limitation is that this method is biased by current knowledge
   \end{enumerate}
\end{itemize}
\end{frame}



\begin{frame}\frametitle{Types of multi-staged analyses}

\begin{figure}
  \subfloat[Genomic variation analysis]{\includegraphics[height=11cm, width=7cm]{figures/multi_staged_1.pdf}} 
  \subfloat[Domain knowledge-guided]{\includegraphics[height=11cm, width=7cm]{figures/multi_staged_2.pdf}}
\end{figure}

\end{frame}


\begin{frame}\frametitle{Genomic variation example}

\begin{figure}
  \includegraphics[height=10cm, width=8cm]{figures/multi_staged_example_1.pdf}
\end{figure}

\end{frame}


\begin{frame}\frametitle{Genomic variation example}

\begin{figure}
  \includegraphics[height=6cm, width=4.5cm]{figures/multi_staged_example_3.jpg}
\end{figure}

\end{frame}


\begin{frame}\frametitle{Genomic variation example}

\begin{itemize}
 \item Mediation analysis
 \item Mendelian randomization (MR)
 \item Conditional Independent test (CIT) 
\end{itemize}

Mediation: \url{https://cran.r-project.org/web/packages/mediation/vignettes/mediation.pdf}
MR: \url{https://cran.r-project.org/web/packages/MendelianRandomization/vignettes/Vignette_MR.pdf}
CIT:\url{https://cran.r-project.org/web/packages/cit/}

\end{frame}


\begin{frame}\frametitle{Mediation analysis}

\begin{figure}
  \includegraphics[height=3.5cm, width=6cm]{figures/mendelian_randomization.png}
\end{figure}

See file {\tt mediation.html}

\end{frame}


\begin{frame}\frametitle{Conditional Independent Test}

\begin{figure}
  \includegraphics[height=6cm, width=7.5cm]{figures/CIT.png}
\end{figure}

\end{frame}


\begin{frame}\frametitle{Domain knowledge-guided}

\begin{figure}
  \includegraphics[height=7.5cm, width=6.5cm]{figures/roadmap.jpg}
\end{figure}

\end{frame}




\begin{frame}\frametitle{The RoadMap Epigenomics}

\begin{itemize}
 \item \textcolor{OliveGreen}{Marking of functional genomic elements}
 \item \textcolor{OliveGreen}{Environmental exposures}
 \item Understanding development and differentiation
 \item Regenerative medicine (stem and iPS cells)
 \item Human disease
 \item \textcolor{OliveGreen}{Interpreting GWAS}
 \item Biomarkers, diagnostics and therapies
 \item Exploring cross-talk between epigenomic mechanisms
\end{itemize}

\end{frame}


\begin{frame}\frametitle{Domain knowledge-guided}

\begin{figure}
  \includegraphics[height=7.5cm, width=6.5cm]{figures/roadmap_utility.jpg}
\end{figure}

\end{frame}



\begin{frame}\frametitle{Domain knowledge-guided}

\begin{figure}
  \includegraphics[height=7.5cm, width=6.5cm]{figures/t2d.jpg}
\end{figure}

\end{frame}



\begin{frame}\frametitle{Domain knowledge-guided}

\begin{figure}
  \includegraphics[height=7.5cm, width=6.5cm]{figures/t2d_roadmap.jpg}
\end{figure}

\end{frame}


\begin{frame}[fragile]\frametitle{Domain knowledge-guided}

Let us assume that we want to extract biological information (eQTL, LD, motifs, ...) from large genomic projects such as ENCODE, the 1000 Genomes Project, Roadmap Epigenomics Project and others on the 15 SNPs obtained in paper \textit{Verweij et al. Identification of 15 novel risk loci for coronary artery disease and genetic risk of recurrent events, atrial fibrillation and heart failure. Sci Rep, 7, Article number: 2761 (2017)}. 
<<haploR>>=
library(haploR)
x <- queryHaploreg(query=c(c("rs11810571", "rs7623687", "rs142695226", 
                             "rs433903", "rs10857147", "rs11723436",
                             "rs35879803", "rs35541991", "rs1351525",
                             "rs11170820", "rs2244608", "rs3832966",
                             "rs33928862", "rs7500448", "rs138120077")))
@

\end{frame}


\begin{frame}[fragile]\frametitle{Domain knowledge-guided}

We can ask, for instance, information about motifs, gwas, gene name and Enhancer histone marks of those SNPs that are in high LD (>0.9). Notice that the name of all possible variables can shown by typing {\tt names(x)}.

<<haploR_2>>=
sel <- as.numeric(x$r2) > 0.9
x2 <- x[sel, c("pos_hg38", "rsID", "Motifs", "gwas",
               "GENCODE_name", "Enhancer_histone_marks")]
x2
@

\end{frame}

\begin{frame}[fragile]\frametitle{Domain knowledge-guided}

eQTLs can also be depict by

<<haploR_3>>=
x3 <- x[, c("rsID", "eQTL")]
x3
@


\end{frame}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Meta-dimensional analysis}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\begin{frame}\frametitle{Meta-dimensional analysis}

\begin{itemize}
\item \textcolor{Gray}{Multi-staged analysis: Involves integrating information using a stepwise or hierarchical analysis approach}
\item \textcolor{OliveGreen}{Meta-dimensional analysis}: Refers to the concept of integrating multiple different data types to build a multivariate model associated with a given outcome
\end{itemize}
\end{frame}


\begin{frame}\frametitle{Types of meta-dimensional analyses}

\begin{figure}
\begin{center}
 \includegraphics[height=7cm, width=6cm]{figures/meta_dimensional.jpg}
\end{center}
\end{figure}

\end{frame}







%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{One omic dataset: dimensionality reduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\begin{frame}\frametitle{One omic dataset: dimensionality reduction}

\begin{figure}
\begin{center}
 \includegraphics[height=5cm, width=5cm]{figures/pca_example.jpg}
\end{center}
\end{figure}

\end{frame}

\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

\begin{itemize}
\item Used to study tissue, cell attributes or cancer signatures with 
regard to abundance of mRNAs, proteins and metabolites
\item Given an omic data set $X$, which is a $n \times p$ matrix, of $n$ individuals and $p$ features
$$ X=(x_1, x_2, \ldots, x_p)$$ 
\item we look for new variables that are linear combinations of the original variables
$f=q_1X_1 + q_2X_2 + \ldots + q_pX_p$  or $f=Xq$ where $q$ are know as loadings.
\item We introduce the restriction that for $i$th component, $q$'s should maximize the variance components of $f$'s
$$ \argmax_{q^i} \text{var}(Xq^i)$$ and $q$'s has to be orthogonal to each other.
\end{itemize}

\end{frame}

\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

<<plot_pca_1, echo=FALSE>>=
set.seed(12345)
y <- rnorm(45, 4, 1)
x <- 2 * y + rnorm(45)
cc <- rep("black", 45)
cc[x>10] <- "red"
cc[c(4,9)] <- "red"
plot(x,y, cex.lab=1.4, cex.axis = 1.3, xlab="GATA3", ylab="XPBD1")
points(x,y, pch=16, col=cc, cex=1.3)
@

\end{frame}

\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

<<plot_pca_2, echo=FALSE>>=
plot(x,y, cex.lab=1.4, cex.axis = 1.3, xlab="GATA3", ylab="XPBD1")
points(x,y, pch=16, col=cc, cex=1.3)
abline(lm(y~x), col="darkgreen", lwd=2)
arrows(6.5, 2.5, 6.5, 3.2, length=0.15, col="darkgreen")
text(6.5, 2.3, "First PCA", col="darkgreen")
@

\end{frame}


\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

<<plot_pca_3, echo=FALSE>>=
plot(x,y, cex.lab=1.4, cex.axis = 1.3, xlab="GATA3", ylab="XPBD1")
points(x,y, pch=16, col=cc, cex=1.3)
abline(lm(y~x), col="darkgreen", lwd=2)
arrows(6.5, 2.5, 6.5, 3.2, length=0.15, col="darkgreen")
text(6.5, 2.3, "First PCA", col="darkgreen")
segments(10, 3.4, 7.6, 5.0, col="darkgreen")
arrows(11.7, 3.6, 10, 3.6, length=0.15, col="darkgreen")
text(12.0, 3.6, "Second PCA", adj=0, col="darkgreen")
@

\end{frame}




\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

<<plot_pca_4, echo=FALSE>>=
ans <- princomp(cbind(y,x))
plot(ans$score, xlab="PCA1", ylab="PCA2", col=cc, pch=16, cex.lab=1.4, cex.axis = 1.3)
abline(h=0, lty=2)
abline(v=0, lty=2)
@

\end{frame}


\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

Data visualization: biplot

<<plot_pca_5, echo=FALSE>>=
plot(ans$score, xlab="PCA1", ylab="PCA2", col=cc, pch=16, cex.lab=1.4, cex.axis = 1.3)
abline(h=0, lty=2)
abline(v=0, lty=2)
arrows(0,0, 3.5,0.2, col="darkblue", length=0.05)
arrows(0,0, 1, -0.1, col="darkblue", length=0.05)
text(3.6, 0.2, "GATA3", col="darkblue", adj=0)
text(1.1, -0.1, "XPBD1", col="darkblue", adj=0)
@

\end{frame}


\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

\begin{itemize}
\item There are other thechniques such as: Principal co-ordinate analysis (PCoA), correspondence analysis (CA) and nonsymetrical correspondence analysis (NSCA).
\pause
\item They can be applied to other types of data (e.g. non-continuous). PCoA can also be applied to binary or count data.
\pause
\item PCA is designed to analyze multi-normal distributed data. If data are skweed, contain extreme outliers, or display nonlinear trends, other methods such as 
Multidimendional Scaling (MDS)
\pause
\item Nonnegative matrix factorization (NMF) and Independent Component Analysis (ICA) are applied when orthogonality or independence acrross components are not hold.
\end{itemize}

\end{frame}


\begin{frame}[fragile]\frametitle{One omic dataset: dimensionality reduction}

 \begin{itemize}
 \item Solving the problem for the $i$-th component $$ \argmax_{q^i} \text{var}(Xq^i)$$ uses SVD decomposition and it requires and inversion step that can be problematic when   $p >> n$
 \item Several extensions based on reguralization step or L-1 penalization (Least Absolute Shrinkage and Selection Operator, LASSO) can be applied
 \item Sparse, penalized and reguralized extensions of PCA and related method have been proposed in omic data analysis, recently (sPCA)
 \end{itemize}
\end{frame}



\begin{frame}{Multi-omic real data}

\begin{itemize}
  \item Data from the Cancer Genome Atlas (TCGA) will be analyzed.
  \item A subset of the TCGA breast cancer study from Nature 2012 publication have been selected.
  \item Data {\tt https://tcga-data.nci.nih.gov/docs/publications/brca\_2012/}.
  \item Available data are: miRNA, miRNAprecursor, RNAseq, Methylation, proteins from a RPPA array, and  GISTIC SNP calls (CNA and LOH). Clinical data are also available.
 \end{itemize}

\end{frame}


\begin{frame}[fragile]\frametitle{Multi-omic real data}

Data can be loaded into {\tt R} by executing:

<<load_data>>=
load("data/breast_TCGA_subset_multi_omic.RData")
summary(breast_multi)
@
\end{frame}



\begin{frame}[fragile]\frametitle{PCA analysis}

<<load_ade4>>=
require(ade4)
dim(breast_multi$RNAseq)
@

<<pca, cache=TRUE>>=
breastPCA<-dudi.pca(breast_multi$RNAseq, 
            scannf=FALSE, nf=5)
@
\end{frame}


\begin{frame}[fragile]\frametitle{PCA analysis}

There is a BioC package called made4 which is a wrapper oround ade4 that 
utilizes Bioconductor data classes (such as {\tt Expression Set} to help 
visualizing the results). They key function ir {\tt ord}. The panels shows
plot of eignevalues, projections of samples, projections of genes and biplot.

<<load_made4>>=
require(made4)
@

<<pca_plot, fig.show='hide'>>=
group<-droplevels(breast_multi$clin$ER.Status)
out <- ord(breast_multi$RNAseq, classvec=group)
plot(out, nlab=3, arraylabels=rep("T", 79))
@
\end{frame}


\begin{frame}[fragile]\frametitle{PCA analysis}

\begin{figure}
\begin{center}
 \includegraphics[width=5.5cm]{figure/pca_plot-1}
\end{center}
\end{figure}
\end{frame}

\begin{frame}[fragile]\frametitle{PCA analysis}

Sample and genes projections can also be plotted using another wrapper around
{\tt ade4} package. By default the top-10 more important genes for each axis
are highlighted.
<<projections_plot, fig.show='hide'>>=
par(mfrow=c(2,1))
plotarrays(out$ord$co, classvec=group)
plotgenes(out, col="blue")
@
\end{frame}


\begin{frame}[fragile]\frametitle{PCA analysis}

\begin{figure}
\begin{center}
 \includegraphics[width=5.5cm]{figure/projections_plot-1}
\end{center}
\end{figure}


\end{frame}


\begin{frame}[fragile]\frametitle{PCA analysis}

A list of variables with higher loadings on axes can be obtained using the following
function

<<list_genes>>=
ax1 <- topgenes(out, axis=1, n=5)
ax2 <- topgenes(out, axis=2, n=5)
cbind(ax1, ax2)
@

\end{frame}



\begin{frame}[fragile]\frametitle{Sparse PCA analysis}

Witten D, Tibshirani R, Hastie T (Biostatistics, 2009, 10(3):515-534) presented a penalized matrix decomposition for computing a rank-K approximation for a matrix based on L1-penalization. The method is also extended to Canonical Correlation Analysis (CCA).

<<sPCA, cache=TRUE>>=
require(PMA)
dd <- t(breast_multi$RNAseq)
sout <- SPC(dd, sumabsv=3, 
            K=2, orth=TRUE)
@

NOTE: {\tt subabsv} argument controls the degree of sparsity. It can be tuned by using {\tt SPC.cv} function

\end{frame}


\begin{frame}[fragile]\frametitle{Sparse PCA analysis}

<<sPCA_out, size='scriptsize'>>=
rownames(sout$u) <- rownames(dd)
rownames(sout$v) <- colnames(dd)
head(sout$u)
head(sout$v)
@

\end{frame}

\begin{frame}[fragile]\frametitle{Sparse PCA analysis}

<<sPCA_plot>>=
plot(sout$u, type="n", xlab="First sPCA", ylab="Second sPCA")
points(sout$u, col=as.numeric(group))
@

\end{frame}


\begin{frame}[fragile]\frametitle{Sparse PCA analysis}

<<sPCA_genes, size='scriptsize'>>=
ss <- sout$v[,1]
ss[ss!=0]
ax1
@

\end{frame}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Integrating two or more omic datasets}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



\begin{frame}\frametitle{One omic dataset: dimensionality reduction}

\begin{figure}
\begin{center}
 \includegraphics[height=5.5cm, width=6cm]{figures/multi_example.jpg}
\end{center}
\end{figure}

\end{frame}


\begin{frame}\frametitle{Integrating two or more omic datasets}

\begin{itemize}
\item There are methods based on dimension reduction techniques: generalized SVD, Co-Inertia Analysis (CIA), sparse or penalized extensions of Partial Least Squares (PLS),
Canonical Correlation (CCA), Multiple Factor Analysis (MFA), Generalized Canonical Correlation (GCA)
\pause
\item There are methods that are based on path modelling: structural equation models (SEM)
\pause
\item Reguralized Generalized Canonical Correlation (RGCA) provides a unified framework for different approaches (Tenenhaus, M. and Tenenhaus, A. Psychometrika, 2011, 76(2):257-284). Other methods are special cases of RGCA.
\pause
\item RGCA integrates a feature selection method, named sparse GCCA (SGCCA) (Tenenhaus, A et al. Biostatistics, 2014, 15(3):569-83).
\end{itemize}

\end{frame}




\begin{frame}\frametitle{Integrating two or more omic datasets}

\begin{itemize}
\item Canonical Correlation can be seen as an extension of PCA for more than two tables $X$ and $Y$
\item The two datasets can be decomposed as:
$$ f =Xp $$ $$ g =Yq $$
where $p$ and $q$ are the loading vectors
\item CCA searches for association or correlations among $X$ and $Y$ by
$$ \argmax_{p^i q^i} \text{cor}(Xp^i Yq^i)$$
for the $i$-th component
\item $Xp^i$ and $Yq^i$ are known as canonical variates and their correlations are the canonical correlations.
\end{itemize}

\end{frame}


\begin{frame}{Integrating two or more omic datasets}

\begin{itemize}
\item ($p >> n$) is an issue. Additionally, there is often presence of multicollinearity within both sets of variables that requires a reguralization step.
\item This may accomplished by addign a ridge penalty, that is, adding a multiple of the identity matrix to the correlation/covariance matrix.
$$ \argmax_{p^i q^i} \text{cor}(Xp^i Yq^i) + \lambda \text{I}$$

\item A sparse solution (filtering the number of variables) is the solution: pCCA, sCCA, CCA-l1, CCA-EN, CCA-group sparse have been used to integrate two omic data.
\end{itemize}
\end{frame}



\begin{frame}\frametitle{Integrating two or more omic datasets}

\begin{figure}
\begin{center}
 \includegraphics[height=3.5cm, width=6cm]{figures/canonical_correlation.jpg}
\end{center}
\end{figure}

$V_y$ and $V_x$ are selected to maximize:

\begin{itemize}
 \item Correlation (CCA)
 \item Squared Covariance (CIA)
       $$ \argmax_{p^i q^i} \text{cov}^2(Xp^i Yq^i)$$
\end{itemize}

\end{frame}

\begin{frame}\frametitle{Canonical Correlation}

\begin{itemize}
\item CCA has been used in omic data
\item The main limitation is that the number of features generally greatly exceeds the number of observations
\item Consequence 1: parameter estimation cannot be applied using standard methods
\item  Consequence 2: Most of markers are having no effect in the canonical axes (e.g. almost all components in a and b are 0)
\item Penalized (sparse) CCA has been proposed
\item Main advantages: More than two tables, easy interpretation (do not need computing p-values nor correcting for multiple comparisons)
\end{itemize}

\end{frame}



\begin{frame}\frametitle{Sparse canonical correlation}

\begin{figure}
\begin{center}
 \includegraphics[height=5cm, width=8cm]{figures/mCCA.png}
\end{center}
\end{figure}

\end{frame}

\begin{frame}{Co-Inertia}

\begin{itemize}

  \item CIA does not requires an inversion step of the covariance matrix; thus, regularization or penalization implementation is not required
  \item CIA can deal with disperse variables
  \item CIA does consider quantitative or qualitative variables
  \item Can weight cases
  \item The method provides the RV coefficient. This is a measure of global similarity between the datasets, and is a number between 0 and 1. The closer it is to 1 the greater the global similarity between the two datasets. 
 \end{itemize}

\end{frame}



\begin{frame}[fragile]\frametitle{Canonical Correlation: gene expression and proteins}

<<data_cc>>=
require(CCA)
df1 <- t(breast_multi$RNAseq)[,1:1000]
df2 <- t(breast_multi$RPPA)
@

<<cc, eval=FALSE>>=
resCC <- cc(df1, df2) 
@


\begin{verbatim}
Error en chol.default(Bmat) : 
  la submatriz de orden 81 no es definida positiva
\end{verbatim}

<<rcc, cache=TRUE>>=
resRCC <- rcc(df1, df2, 0.2, 0.1) 
@

<<regul_estim, eval=FALSE>>=
regul <- estim.regul(df1, df2)
resRCC2 <- rcc(df1, df2, regul$lambda1, regul$lambda2)
@

<<plot_rcc, fig.show='hide'>>=
plt.cc(resRCC)
@

\end{frame}


\begin{frame}[fragile]\frametitle{Canonical Correlation: gene expression and proteins}

<<plot_rcc_out, fig.width=12, echo=FALSE>>=
plt.cc(resRCC)
@

\end{frame}




\begin{frame}[fragile]\frametitle{Canonical Correlation: gene expression and proteins}

<<multiCCA>>=
require(PMA)
ddlist <- list(df1, df2)
perm.out <- MultiCCA.permute(ddlist, 
                             type=c("standard", "standard"),
                             trace=FALSE) 
resMultiCCA <- MultiCCA(ddlist,  
                        penalty=perm.out$bestpenalties, 
                        ws=perm.out$ws.init, 
                        type=c("standard", "standard"), 
                        ncomponents=1, trace=FALSE, standardize=TRUE)
@

NOTE: setting {\tt type} equal to "ordered" allows to consider that features are correlated (e.g. genomic regions) 
\end{frame}


\begin{frame}[fragile]\frametitle{Canonical Correlation: gene expression and proteins}

<<multiCCA_out, size='scriptsize'>>=
rownames(resMultiCCA$ws[[1]]) <- colnames(df1)
rownames(resMultiCCA$ws[[2]]) <- colnames(df2)
head(resMultiCCA$ws[[1]])
head(resMultiCCA$ws[[2]])
@
\end{frame}





\begin{frame}[fragile]\frametitle{Co-inertia: gene expression and proteins}

Let us load required packages (coinertia and multiple coinertia)
<<getcia>>=
library(made4)
library(omicade4)
@

<<cia, cache=TRUE>>=
resCIA <- cia(breast_multi$RNAseq, breast_multi$RPPA)
@

<<plot_cia, fig.show='hide'>>=
plot(resCIA, classvec=group, nlab=3, clab=0, cpoint=3 )
@

\end{frame}



\begin{frame}[fragile]\frametitle{Co-inertia: gene expression and proteins}

\begin{figure}
\begin{center}
 \includegraphics[width=6cm]{figure/plot_cia-1}
\end{center}
\end{figure}

\end{frame}


\begin{frame}[fragile]\frametitle{Co-inertia: gene expression and proteins}

Top-5 features of the first axis (positive side) can be retrieve by

\footnotesize 
<<top_features>>=
topVar(resCIA, axis=1, topN=5, end="positive")
@

\end{frame}



\begin{frame}[fragile]\frametitle{Co-inertia: gene expression and proteins}

Top-5 features of the first axis (negative side) can be retrieve by

\footnotesize 
<<top_features_neg>>=
topVar(resCIA, axis=1, topN=5, end="negative")
@

\end{frame}



\begin{frame}[fragile]\frametitle{More than two tables}


<<mcia, cache=TRUE>>=
resMCIA <- mcia( breast_multi[ c(1,3,4,5,6,7) ] )
@


<<plot_mcia, fig.show='hide'>>=
plot(resMCIA, axes=1:2, sample.lab=FALSE, sample.legend=FALSE, 
     phenovec=group, gene.nlab=2, 
     df.color=c("cyan", "magenta", "red4", "brown","yellow", "orange"),
     df.pch=2:7)
@
\end{frame}


\begin{frame}[fragile]\frametitle{More than two tables}

\begin{figure}
\begin{center}
 \includegraphics[width=5.5cm]{figure/plot_mcia-1}
\end{center}
\end{figure}

\end{frame}

\begin{frame}[fragile]\frametitle{More than two tables}

Top-5 features of the first axis (positive side) can be retrieve by

\scriptsize 
<<top_features_m>>=
topVar(resMCIA, end="positive", axis=1, topN=5)
@

\end{frame}



\begin{frame}[fragile]\frametitle{More than two tables}

<<plot_eigen, fig.show='hide'>>=
plot(resMCIA$mcoa$cov2,  xlab = "pseudoeig 1", 
     ylab = "pseudoeig 2", pch=19, col="red")
text(resMCIA$mcoa$cov2, labels=rownames(resMCIA$mcoa$cov2), 
     cex=1.4, adj=0)
@

\begin{figure}
\begin{center}
 \includegraphics[width=5.5cm]{figure/plot_eigen-1}
\end{center}
\end{figure}

\end{frame}


\begin{frame}\frametitle{Take home messages}

\begin{description}[labelwidth=\widthof{\textbf{Canonical correlation}}]
 \item[Correlation] Y = X
 \item[Multiple correlation] $Y = X_1 X_2 X_3 \cdots X_k$
 \item[Canonical correlation] $Y_1 Y_2 Y_3 \cdots Y_j = X_1 X_2 X_3 \cdots X_k$
\end{description}

\begin{itemize}
  \item Co-inertia analysis (CIA) is similar to CCA but it optimizes the squared covariance between the eigenvectors while CC optimizes
        the correlation. 
  \item CIA can be applied to datasets where the number of variables (genes) far exceeds the number of samples (arrays) such is the case in several omic data, while CCA requires a regularized version to be implemented. 
  \item Sparse and reguralized methods requires a tuning parameter. This makes these methods computing demanding.
\end{itemize}


\end{frame}



\begin{frame}\frametitle{Take home messages}
\begin{itemize}
  \item Multivariate methods are purely descriptive methods that do not test a hypothesis to generate a p-value.
  \item They are not optimized for variable of biomarkers discovery, though the introduction of sparsity in variable loadings may help in the selection of variables forn downstream analyses.
  \item Number of variables in omic data is a challenge to tradicional visualization tools. New {\tt R} packages including {\tt ggord} are being developed to address this issue.
  \item Dynamic visualization is possible using {\tt ggvis}, {\tt ploty}, {\tt explor} and other packages.
  \item Projection in the same space of variable annotation (GO or Reactome) may help to determine gene sets or patwhays associated with our traits.

\end{itemize}
\end{frame}


\section{Recommended lectures}

\begin{frame}\frametitle{Recommended lectures}

\footnotesize

\begin{itemize}
\item Richmond et al. (2016). Challenges and novel approaches for investigating molecular mediation. Hum Mol Genet, 25(R2):R149-R156.
\item Millstein et al. (2009). Disentangling molecular relationships with a causal inference test. BMC Bioinf, 10:23.
\item Ebrahim and Smith (2008). Mendelian randomization: can genetic epidemiology help redress the failure of observational epidemiology?. Hum Genet, 123:15-33.
\item Liu et al. (2013). Epifenome-wide association data implicate DNA methylation as an intermediary of genetic risk in rheumatoid arthritis. Nat Biotech, 31(2):142-148.
\item Voight et al. (2012). Plasma HDL cholesterol and risk of myocardial infarction: a mendelian randomisation study. Lancet, 380(9841): 572-580.
\item Meng et al. (2014). A multivariate approach to the integration of multi-omics datasets. BMC Bioinf, 15:162.
\item Witten et al. (2009). A Penalized Matrix Decomposition, with Applications to Sparse Principal Components and Canonical Correlation Analysis. Biostatistics, 10(3):515-34.
\item Meng et al (2016). Dimension reduction techniques for the integrative analysis of multi-omics data. Brief Bioinform, 17(4): 628-641. 
\end{itemize}
\end{frame}



\end{document}















